### ISSUE TITLE
Refactor implementation - AbsolutePositionalEncoding class

### ISSUE THREAD

**Issue by soran-ghaderi** [tensorops/TransformerX#71]
- The sin and cos functions are being called from both numpy and tensorflow. This can cause errors if the types are not properly converted between the two libraries. In this case, it's better to just use the tf.sin and tf.cos functions. Also, less conversion leads to less time and space complexity. 

- The P matrix is created using numpy and then converted to a tensorflow tensor using tf.convert_to_tensor. This is unnecessary, as the P matrix can be created directly as a tensorflow tensor using tf.sin and tf.cos.

- The tf.keras.layers.Dropout layer is used to apply dropout to the entire X tensor, including the positional encoding. This is not ideal, as the positional encoding is meant to be fixed and not subject to dropout. Instead, the dropout layer should be applied only to the input features.
